flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamOneInputProcessor.java

```diff
@@ -188,26 +132,59 @@ public void close() throws IOException {
 		input.close();
 	}
 
-	private class ForwardingValveOutputHandler implements StatusWatermarkValve.ValveOutputHandler {
+	/**
+	 * The network data output implementation used for processing stream elements
+	 * from {@link StreamTaskNetworkInput} in one input processor.
+	 */
+	private class StreamTaskNetworkOutput implements DataOutput<IN> {
```

Comment:
Shouldn't the name be more generic and not connected with `Network`? Isn't the same class going to be used for the source reader?

Maybe if we are not sure about that, we can leave this as it is now and resolve it once we add `SourceReader` support?

Reply:
It is hard to say. This implementation is for `OneInputStreamOperator` and I am not sure the new source reader could also couple with `OneInputStreamOperator`.

We can keep it as now and see how the formal `SourceReader` works.